scripts/json2synapse.py

```diff
@@ -0,0 +1,217 @@
+neuro = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/neuro.json'
+neurofibromatosis = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/neurofibromatosis.json'
+ngs = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/ngs.json'
+tool = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/tool.json'
+toolExtended = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/toolExtended.json'
+
+paths = [analysis, sageCommunity, array, cancer, dhart, experimentalData, genie, neuro, neurofibromatosis, ngs, tool, toolExtended]
+names = ['analysis', 'sageCommunity', 'array', 'cancer', 'dhart', 'experimentalData', 'genie', 'neuro', 'neurofibromatosis', 'ngs', 'tool', 'toolExtended']
+
+
+all_modules = []
+
+tableSynId = "syn10242922"
+key = ["key", "value"]
+
+
+def json2flatten(path, module):
+    # fetch and read raw json objects from its' github url and decode the json object to its raw format
+    json_record = pandas.read_json(path)
+
+    # grab annotations with empty enumValue lists i.e. don't require normalization and structure their schema
+    empty_vals = json_record.loc[json_record.enumValues.str.len() == 0]
+    empty_vals = empty_vals.drop('enumValues', axis=1)
+
+    empty_vals['enumValues_description'] = ""
+    empty_vals['enumValues_source'] = ""
+    empty_vals['enumValues_value'] = ""
+    empty_vals['module'] = module
+
+    empty_vals.set_index(empty_vals['name'], inplace=True)
+
+    # for each value list object
+    flatten_vals = []
+    json_record = json_record.loc[json_record.enumValues.str.len() > 0]
+    json_record.reset_index(inplace=True)
+
+    for i, jsn in enumerate(json_record['enumValues']):
+        df = pandas.io.json.json_normalize(json_record['enumValues'][i])
+        # re-name columns to match table on synapse schema
+        df = df.rename(columns={'value': 'enumValues_value', 'description': 'enumValues_description',
+                                'source': 'enumValues_source'})
+
+        # grab key information in its row, expand it by values dimention and append its key-columns to flattened values
+        rows = json_record.loc[i:i, json_record.columns != 'enumValues']
+        repeats = pandas.concat([rows] * len(df.index))
+        repeats.set_index(df.index, inplace=True)
+        flatten_df = pandas.concat([repeats, df], axis=1)
+        # append module category / annotating the annotations for module filtering
+        flatten_df['module'] = module
+        flatten_df.set_index(flatten_df['name'], inplace=True)
+
+        flatten_vals.append(flatten_df)
+
+    flatten_vals.append(empty_vals)
+
+    return flatten_vals
+
+
+def rightJoin(df1, df2, key, suffixes):
```

Comment:
This is the output of pandas.merge with a rightjoin, 
I change the Exist column values 

```python 
In [177]: pandas.merge(df1, df2, on=['key1', 'key2', 'key3'], how='right', indicator='Exist', suffixes=('_cur', '_new'))
Out[]:
   col1_cur  col2_cur       key1 key2 key3  col1_new  col2_new       Exist
0       1.0       2.0          a    x    (         1         3        both
1       1.0       2.0          b    y    *         1         2        both
2       NaN       NaN  change me    z    )         1         3  right_only
```